### PR TITLE
Updated CONTRIBUTING.md for setup of venv and standard pip install

### LINKED ISSUE #385: `SetuptoolsDeprecationWarning`: `setup.py` install is deprecated. Use build and pip and other standards-based tools.
Following the issue, I resolved at Keras-NLP [#112](https://github.com/keras-team/keras-nlp/issues/112).

We need to remove the `setup.py` instruction for development environment setup, and recommend to add virtual environment directory.

### PR DESCRIPTION
**Pull request by adhadse**
Resolves #385.
@bhack, @LukeWood this PR is ready for review!

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -54,16 +54,20 @@ Once the pull request is approved, a team member will take care of merging.
...
+install dependencies.
 
 You can achieve this by running the following commands:
 
 ```shell
 gh repo fork keras-team/keras-cv --clone --remote
 cd keras-cv
-pip install ".[tests]"
-python setup.py develop
+python -m venv ~/keras-cv-venv
```
Comment (bhack) [inline .github/CONTRIBUTING.md:68]: I have a question. I this required when we work with our [devcontainer/codespace](https://github.com/keras-team/keras-cv/tree/master/.devcontainer)? If not I suppose that we could add a notice

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -54,16 +54,20 @@ Once the pull request is approved, a team member will take care of merging.
 
 ## Setup environment
 
-Setting up your KerasCV development environment requires you to fork the KerasCV repository,
-clone the repository, install dependencies, and execute `python setup.py develop`.
+Python 3.7 or later is required.
+
+Setting up your KerasNLP development environment requires you to fork the
```
Comment (bhack) [inline .github/CONTRIBUTING.md:59]: KerasCV

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -54,16 +54,20 @@ Once the pull request is approved, a team member will take care of merging.
...
+KerasNLP repository, clone the repository, create a virtual environment, and
```
Comment (bhack) [inline .github/CONTRIBUTING.md:60]: Ditto

File: .gitignore [hunk 4]
```diff
@@ -13,3 +13,7 @@ __pycache__/
 .vscode/
 .devcontainer/
 .coverage
+
+#Virtual environment and IDE files
+venv/
```
Comment (bhack) [inline .gitignore:18]: Why we need this if we suggest to use `~/keras-cv-venv`?

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (adhadse) [reply .github/CONTRIBUTING.md:68]: I suppose Docker is completely independent of python virtual env. I didn't understand what kind of notice?

File: .gitignore (hunk 4 above)
Comment (adhadse) [reply .gitignore:18]: Often time IDEs don't create `venv` in the root directory, especially if you're working on Windows. And since the default name goes with `venv` I added it (actually I do have `venv` directory in my local setup too).

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (bhack) [reply .github/CONTRIBUTING.md:68]: Oh I did not remember that we still don't have an section to use `[devcontainer/codespace](https://github.com/keras-team/keras-cv/tree/master/.devcontainer)`. 
Probably we will add this later or we can open a separate ticket to track this.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -54,16 +54,20 @@ Once the pull request is approved, a team member will take care of merging.
...
+install dependencies.
 
 You can achieve this by running the following commands:
 
 ```shell
 gh repo fork keras-team/keras-cv --clone --remote
 cd keras-cv
-pip install ".[tests]"
-python setup.py develop
+python -m venv ~/keras-cv-venv
```
Comment (LukeWood) [inline .github/CONTRIBUTING.md:68]: can we preserve the original instructions as well and offer this as an alternative?

**Comment by LukeWood** [comment]
Thanks for the PR, but I wonder if this is best offered as an alternative to users.  Perhaps in an addendum at the end.

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (adhadse) [reply .github/CONTRIBUTING.md:68]: Sure!

**Comment by adhadse** [comment]
Our job is to always make sure to streamline the process to get the developer into action. Having a `venv` makes sure our project does not cause issues with other projects. 
And as always it's just a recommendation.

I'll recommend the setup of the virtual environment as an alternative down below the setup commands and update this PR.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (adhadse) [reply .github/CONTRIBUTING.md:68]: Yeah 👍

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -54,20 +54,28 @@ Once the pull request is approved, a team member will take care of merging.
 
 ## Setup environment
 
-Setting up your KerasCV development environment requires you to fork the KerasCV repository,
```
Comment (LukeWood) [inline .github/CONTRIBUTING.md:57]: Can you revert the change to this line

**Comment by LukeWood** [comment]
This removes critical information from the docs and has been stale for quite some time.

Please feel free to re-open this PR and address the comments; for now closing.
